src/datasources/asset/types.ts

```diff
@@ -0,0 +1,140 @@
+  supportsReset: boolean,
+  id: string,
+  location: AssetLocationModel,
+  calibrationStatus: 'OK' | 'APPROACHING_RECOMMENDED_DUE_DATE' | 'PAST_RECOMMENDED_DUE_DATE',
+  isSystemController: boolean,
+  externalCalibration?: ExternalCalibrationModel,
+  discoveryType: 'MANUAL' | 'AUTOMATIC',
+  properties: Record<string, string>,
+  keywords: string[],
+  lastUpdatedTimestamp: string,
+}
+
+export interface AssetPresenceWithSystemConnectionModel {
+  assetPresence: AssetPresence,
+  systemConnection?: SystemConnection
+}
+
+export interface AssetLocationModel {
+  minionId: string,
+  parent: string,
+  resourceUri: string,
+  slotNumber: number,
+  systemName: string,
+  state: AssetPresenceWithSystemConnectionModel
+}
+
+export interface ExternalCalibrationModel {
+  temperatureSensors: any[],
+  isLimited?: boolean,
+  date: string,
+  recommendedInterval: number,
+  nextRecommendedDate: string,
+  nextCustomDueDate?: string,
+  comments: string,
+  entryType: ExternalCalibrationEntryType,
+  operator: ExternalCalibrationOperatorModel
+}
+
+export interface ExternalCalibrationOperatorModel {
+  displayName: string;
+  userId: string
+}
+
+export enum AssetFilterProperties {
+  AssetIdentifier = 'AssetIdentifier',
+  SerialNumber = 'SerialNumber',
+  ModelName = 'ModelName',
+  VendorName = 'VendorName',
+  VendorNumber = 'VendorNumber',
+  AssetName = 'AssetName',
+  FirmwareVersion = 'FirmwareVersion',
+  HardwareVersion = 'HardwareVersion',
+  BusType = 'BusType',
+  IsNIAsset = 'IsNIAsset',
+  Keywords = 'Keywords',
+  Properties = 'Properties',
+  LocationMinionId = 'Location.MinionId',
+  LocationSystemName = 'Location.SystemName',
+  LocationSlotNumber = 'Location.SlotNumber',
+  LocationAssetStateSystemConnection = 'Location.AssetState.SystemConnection',
+  LocationAssetStateAssetPresence = 'Location.AssetState.AssetPresence',
+  SupportsSelfCalibration = 'SupportsSelfCalibration',
+  SelfCalibrationCalibrationDate = 'SelfCalibration.CalibrationDate',
+  SupportsExternalCalibration = 'SupportsExternalCalibration',
+  CustomCalibrationInterval = 'CustomCalibrationInterval',
+  CalibrationStatus = 'CalibrationStatus',
+  ExternalCalibrationCalibrationDate = 'ExternalCalibration.CalibrationDate',
+  ExternalCalibrationNextRecommendedDate = 'ExternalCalibration.NextRecommendedDate',
+  ExternalCalibrationRecommendedInterval = 'ExternalCalibration.RecommendedInterval',
+  ExternalCalibrationComments = 'ExternalCalibration.Comments',
+  ExternalCalibrationIsLimited = 'ExternalCalibration.IsLimited',
+  ExternalCalibrationOperatorDisplayName = 'ExternalCalibration.Operator.DisplayName',
+  IsSystemController = 'IsSystemController'
+}
+
+export enum AssetPresence {
+  INITIALIZING = "INITIALIZING",
+  UNKNOWN = "UNKNOWN",
+  NOT_PRESENT = "NOT_PRESENT",
+  PRESENT = "PRESENT"
+}
+
+export enum ExternalCalibrationEntryType {
+  AUTOMATIC = 'AUTOMATIC',
+  MANUAL = 'MANUAL'
+}
+
+export enum SystemConnection {
+  APPROVED = "APPROVED",
```

Comment:
In SLE, system connection returned by asset has only 2 values:
DISCONNECTED
CONNECTED

Reply:
will add interface for SLE and merge them like TagWithValue interface for SLS, SLE and SystemLink Cloud

Reply:
see below comment

Reply:
So this is something that needs work with both SLS and SLE, reason you keep old enum values as well?

Reply:
retained both values ​​to support SLS and SLE, added comment